File: javaAPI/src/main/java/com/kotlinjavademo/javaAPI/controllers/VehicleController.java
package com.kotlinjavademo.javaAPI.controllers;

import com.kotlinjavademo.javaAPI.models.Vehicle;
import com.kotlinjavademo.javaAPI.services.VehicleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/vehicles")
public class VehicleController {
    @Autowired
    private VehicleService vehicleService;

    @GetMapping
    public List<Vehicle> getAllVehicles() {
        return vehicleService.retrieveAllVehicles();
    }
    @GetMapping("/{registration}")
    public Vehicle getVehicleByRegistration(@PathVariable String registration) {
        var vehicle = vehicleService.getVehicleByRegistration(registration);
        return vehicle.isPresent() ? vehicle.get() : null;
    }

    @PostMapping("/add")
    public Vehicle addVehicle(@RequestBody Vehicle vehicle) {
        return vehicleService.addVehicle(vehicle);
    }

    @PostMapping("/update")
    public Vehicle updateVehicle(@RequestBody Vehicle vehicle) {
        var updatedVehicle = vehicleService.updateVehicle(vehicle);
        return updatedVehicle.isPresent() ? updatedVehicle.get() : null;
    }
}

File: kotlin/src/main/kotlin/com/kotlinjavademo/kotlin/controllers/VehicleController.kt
package com.kotlinjavademo.kotlin.controllers

import com.kotlinjavademo.kotlin.models.Vehicle
import com.kotlinjavademo.kotlin.services.VehicleService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/vehicles")
class VehicleController (
        @Autowired
        private val vehicleService: VehicleService
) {
    @GetMapping
    fun getAllVehicles() = vehicleService.retrieveAllVehicles()

    @GetMapping("/{registration}")
    fun getVehicleByRegistration(@PathVariable registration: String) : Vehicle? = vehicleService.getVehicleByRegistration(registration)

    @PostMapping("/add")
    fun addVehicle(@RequestBody vehicle: Vehicle) : Vehicle = vehicleService.addVehicle(vehicle)

    @PostMapping("/update")
    fun updateVehicle(@RequestBody vehicle: Vehicle) : Vehicle? = vehicleService.updateVehicle(vehicle)
}

